src/ppx/mipmap.cpp

```diff
@@ -72,8 +82,14 @@ Mipmap::Mipmap(uint32_t width, uint32_t height, Bitmap::Format format, uint32_t
         return;
     }
 
-    mData.resize(dataSize);
-    if (mData.size() != dataSize) {
+    // Choose between static pool use and internal data.
+    // NOTE: This is designed for single-threaded use ONLY!
+    // This will need locks if the consuming paths ever become multi-threaded.
+    std::vector<char>& targetData = mUseStaticPool ? mStaticData : mData;
+    if (targetData.size() < dataSize) {
+        targetData.resize(dataSize);
+    }
+    if (targetData.size() < dataSize) {
```

Comment:
Probably not guranteed, it's just an allocation underneath. We haven't done anything big enough to hit any memory limits. Should probably have an additional assert tho, the return isn't really meaningful since we don't catch exceptions.

Reply:
yeah, I was just duplicating the existing logic, but I don't think it's correct. Looking online, there isn't defined behavior for out of memory issues, but generally a bad_alloc exception should be thrown. Some other implementations simply overrun memory. But there's no definition that indicates a smaller size than requested would be returned.

I think this is one of those 'it's safe to just assume it gives you the memory you ask for, because otherwise it'll just crash' situations. Will remove this logic.

Reply:
Yeah, I cannot find exact rule saying if it should always except or not.  It says there are no effect if an exception is thrown, but not if the allocation can silently fails. Seems like implems I've seen do throw with default allocators, so would agree removing the check should be enough.